### PR TITLE
Pin K8s version in Kind config

### PR DESCRIPTION
**Pull request by mzur**
The Helm chart of v0.8 requires K8s <1.22.0 which will break with the most recent version of kindest/node.

References: https://github.com/reanahub/reana/blob/ed48d7eedf92ff64683a2436c890aa49df70f056/helm/reana/Chart.yaml#L31

### CODE REVIEW

File: etc/kind-localhost-30443.yaml [hunk 1]
```diff
@@ -12,3 +12,4 @@ nodes:
           kubeletExtraArgs:
             node-labels: "ingress-ready=true"
     role: control-plane
+    image: kindest/node:v1.21.12@sha256:f316b33dd88f8196379f38feb80545ef3ed44d9197dca1bfd48bcb1583210207
```
Comment (tiborsimko) [inline etc/kind-localhost-30443.yaml:15]: Hi @mzur, this fix would be OK for Kind 0.14, but will it work well with older Kind versions? Have you tried?

Note that I suggested to use `kind create ... --image=kindest/node:v1.21.12` without the SHA1 part in https://github.com/reanahub/reana/issues/647#issuecomment-1161689205 simply to be more welcoming to older Kind versions, such as 0.13, which may perhaps prefer older SHA1 builds.

In other words, I'm concerned that if we force a particular SHA1 image version here in YAML, it may cause troubles for people using older Kind clients. Especially if somebody uses Kind 0.11 still, they will get Kubernetes 1.21 out of the box, so they wouldn't really need to change anything in the YAML template -- and if we do force this SHA1 here, perhaps their deployment may not work? (To be tested.)

TL;DR if this change could cause troubles for people using older Kind versions, I would rather leave Kubernetes version pinning from the YAML template, and work with this on the CLI level, advising people to install 1.21 depending on which Kind version they use.

Comment (mzur) [reply etc/kind-localhost-30443.yaml:15]: I'm no expert with K8s/Kind and was following the [documentation](https://kind.sigs.k8s.io/docs/user/quick-start/#setting-kubernetes-version) that recommended to specify the checksum. 

I totally missed https://github.com/reanahub/reana/issues/647 which is about the exact issue that I wanted to solve with this PR, sorry. I agree that this can also be fixed with an update of the documentation to pin the K8s version. Please feel free to close this PR.
